fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -214,6 +214,185 @@ public List<JsonObject> queryDataTable(String datatable, final String columnFilt
 
         return results;
     }
+    @Override
+    public Page<JsonObject> queryDataTableSearch(PagedRequest<DatatableSearchRequest> searchRequest) {
+
+        Objects.requireNonNull(searchRequest, "Search Request Should not be Empty");
+        Optional<DatatableSearchRequest> request = searchRequest.getRequest();
+        List<String> resultColumns = request.map(DatatableSearchRequest::getResultColumns).orElse(Collections.emptyList());
+        List<ColumnFilter> columnFilters = request.map(DatatableSearchRequest::getColumnFilters).orElse(Collections.emptyList());
+        String datatable = request.map(DatatableSearchRequest::getDatatable).orElse(null);
+        Pageable pageable = searchRequest.toPageable();
+
+        this.context.authenticatedUser().validateHasDatatableReadPermission(datatable);
+
+        List<JsonObject> results = new ArrayList<>();
+        AtomicInteger totalElements = new AtomicInteger(0);
+        if (CollectionUtils.isNotEmpty(columnFilters) && CollectionUtils.isNotEmpty(resultColumns)) {
+
+            String resultColumn = resultColumns.stream().collect(Collectors.joining(","));
```

Comment:
`resultColumns` is a list of the column names. Having `resultColumn` as a variable name for it is misleading...